godtools/Managers/DownloadResources/DownloadedResourceManager.swift

```diff
@@ -87,6 +97,40 @@ class DownloadedResourceManager: GTDataManager {
         saveToDisk()
     }
     
+    private func translationShouldBeSaved(languageId: String, resourceId: String, version: Int16) -> Bool {
+        let context = NSManagedObjectContext.mr_default()
+        
+        let predicate = NSPredicate(format: "language.remoteId = %@ AND downloadedResource.remoteId = %@",
+                                    languageId,
+                                    resourceId)
+        
+        guard let existingTranslations: [Translation] = Translation.mr_findAll(with: predicate, in: context) as? [Translation] else {
+            // default to saving if in doubt.
+            return true
+        }
+        
+        let latestTranslation = existingTranslations.max(by: {$0.version < $1.version})
+        
+        return latestTranslation == nil || version > latestTranslation!.version
+    }
+    
+    private func purgeTranslationsOlderThan(_ translation: Translation) {
+        let context = NSManagedObjectContext.mr_default()
```

Comment:
All of this is fine, just was thinking for the future we can break all of this in some other methods (like getting the context, or making the predicates, or blocks for when doing the operation) so new implementations will be easier